Enabling the ability to change team area path, making item Id field a link and making datagrid rows with alternate colors

ranm-msft opened the pull request:
Making Id field a link
making DataGrid rows alternate colors
Note: since mod field is applied on all properties, can't set it to false if user changed area path twice (i.e. back to original team)

ranm-msft committed:
Enabling the ability to rename team area path
Making Id field a link
making DataGrid rows alternate colors
Note: since mod field is applied on all properties, can't set it to false if user changed area path twice (i.e. back to original team)

chendrixson commented on Planomatic/DeliverablesPage.xaml.cs:
```diff
@@ -81,7 +81,27 @@ private void OpenItem_Click(object sender, RoutedEventArgs e)
+            {
+                myApp().UpdateStatus($"Team Area path changed, was: {prevTeam}, now: {newTeam}");
+
+                selectedDeliverableItem.AreaPath = selectedDeliverableItem.AreaPath.Replace(
```
If the area path had a sub path under it (as some teams do use), then the update would fail because the new area path with the substituted team wouldn't exist.  Verified this is the case, and that we give a nice status message that the item has failed.

ranm-msft replied:
Fixed. Also i missed a scenario 
there could be 3 scenarios:
1. item path is under one of the paths in the config file
2. item path is under a sub node of one of the paths in the config file (this is the bug i fixed)
3. item path is under a node that is not part of the paths in the config file (i.e. <unmapped>) 
for the 3rd case, we first need to add the 'unmapped' option to the combobbox (to be able to show it in the UI) and ignore if user select it. then if the user wants to map it to a different node, we allow it. Note: we assume that all nodes are under myConfig().RootNode.

chendrixson commented on Planomatic/DeliverablesPage.xaml:
```diff
@@ -52,12 +52,38 @@
-        <DataGrid ItemsSource="{Binding AllItemsView}" AutoGenerateColumns="False" CanUserAddRows="False" x:Name="AllDeliverableList" CanUserSortColumns="False">
+        <DataGrid ItemsSource="{Binding AllItemsView}" AutoGenerateColumns="False"
+                  CanUserAddRows="False" x:Name="AllDeliverableList" CanUserSortColumns="False"
+                  AlternationCount ="2" RowBackground="LightCyan" HorizontalScrollBarVisibility="Visible">
```
Should use a theme color here, like this: {DynamicResource MaterialDesignToolBarBackground}, that looks pretty good here.

I also noticed that with your change only the alternate (cyan in your code) rows have a highlight when you mouse over.  We could probably remove the hover highlight entirely, or fix so it always works.

ranm-msft replied:
Fixed the alternate rows so that both row types are being highlighted.
I prefer to set the style myself - feel free to change :-)

chendrixson reviewed (changes requested):
Thanks for the update, overall works well!  Just a couple comments before accepting the PR.

ranm-msft committed:
Fixed the alternate rows so that both row types are being highlighted.

For Area Path, there could be 3 scenarios:
item path is under one of the paths in the config file
item path is under a sub node of one of the paths in the config file (this is the bug i fixed)
item path is under a node that is not part of the paths in the config file (i.e. )
for the 3rd case, we first need to add the 'unmapped' option to the combobbox (to be able to show it in the UI) and ignore if user select it. then if the user wants to map it to a different node, we allow it. Note: we assume that all nodes are under myConfig().RootNode.

ranm-msft committed:
Added team root area path so we can differentiate form root node.
also, we only change the path if it was modified.
